src/clib/ct.cpp

```diff
@@ -312,7 +316,11 @@ extern "C" {
     size_t thermo_speciesIndex(int n, const char* nm)
     {
         try {
-            return ThermoCabinet::item(n).speciesIndex(nm);
+             size_t k = ThermoCabinet::item(n).speciesIndex(nm);
+             if (k == npos) {
+                ThermoCabinet::item(n).checkSpeciesIndex(k);
+             }
+             return k;
```

Comment:
I think this is getting two things a bit mixed together. The C++ method `speciesIndex` returns `npos` if the named species does not exist. This is a pattern that's used outside of Cantera, for example the Python `str.find()` method returns -1 if the specified substring is not found.

I think it does make sense to change this function to indicate an error if the species is not found. However, this is not the purpose of `checkSpeciesIndex` -- that function is to check if an *input* species index is out of bounds. Instead, you should just throw a `CanteraError` here with the relevant message (which will then immediately be caught by the `catch (...)` block).